### PR TITLE
Fix #6310 - Changing products 'this item has weight' using 'Update Attributes' is not possible

### LINKED ISSUE #6310: Changing products 'this item has weight' using 'Update Attributes' is not possible
### Preconditions
1. Magento CE 2.1.0
2. Have some virtual products in your catalogue
### Steps to reproduce
1. Select a couple of virtual products in the product grid in the backend
2. Choose 'Update Attributes' in the 'Actions' dropdown
3. Search for the weight field, you'll find it
4. Search for the 'This item has weight' or 'This item has no weight' dropdown, you won't find it
### Expected result
1. Expected to find the dropdown with the choices 'This item has weight' or 'This item has no weight'
### Actual result
1. No dropdown with choice 'This item has weight' or 'This item has no weight' found

I wanted to change a couple of 100 products which were Virtual Products because the weight setting was set to 'This item has no weight' to Simple Products using Update Attributes.
But this currently doesn't work, I have to open all those products manually one by one to change them to a Simple Product which is no fun :(

### PR DESCRIPTION
**Pull request by Bartlomiejsz**
### Description (*)

This is fix for #6310

### Related Pull Requests
https://github.com/magento/partners-magento2ee/pull/150

### Fixed Issues (if relevant)

1. magento/magento2#6310: Changing products 'this item has weight' using 'Update Attributes' is not possible

### Manual testing scenarios (*)

1. Select a couple of virtual products in the product grid in the backend
2. Choose 'Update Attributes' in the 'Actions' dropdown
3. Search for the weight field, you'll find it
4. Search for the 'Does this has a weight?', you should be able to find it and use to change multiple products type at once

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by Bartlomiejsz** [comment]
@magento run tests

**Comment by Bartlomiejsz** [comment]
@magento run all tests

**Comment by Bartlomiejsz** [comment]
@magento run all tests

**Comment by Bartlomiejsz** [comment]
@magento run all tests

**Comment by Bartlomiejsz** [comment]
@magento run all tests

**Comment by sivaschenko** [comment]
@magento run all tests

**Review by lenaorobei — DISMISSED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-7151](https://jira.corp.magento.com/browse/ENGCOM-7151) has been created to process this Pull Request

**Review by engcom-Bravo — CHANGES_REQUESTED** [review]
After the Weight attribute has been changed to **Yes** and after ```cron:run```, the Weight successfully changed to _The item has weight_ :+1: 
PR_7151_has weight

But, the **Yes** radio button under _"Does the item have weight?"_ is the default selected. Please change the radio button **No** to be default
PR_7151_default

Expected
PR_7151_default_expected

**Comment by Bartlomiejsz** [comment]
Hi @engcom-Bravo, done

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-7151](https://jira.corp.magento.com/browse/ENGCOM-7151) has been created to process this Pull Request

**Comment by engcom-Bravo** [comment]
:heavy_check_mark: QA Passed

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run all tests

File: app/code/Magento/Catalog/Block/Adminhtml/Product/Helper/Form/Weight.php [hunk 1]
```diff
@@ -84,10 +84,6 @@ public function __construct(
      */
     public function getElementHtml()
     {
-        if (!$this->getForm()->getDataObject()->getTypeInstance()->hasWeight()) {
-            $this->weightSwitcher->setValue(WeightResolver::HAS_NO_WEIGHT);
-        }
-
```
Comment (engcom-Echo) [inline app/code/Magento/Catalog/Block/Adminhtml/Product/Helper/Form/Weight.php]: I think in this case you need to do the check opposite.

>  if ($this->getForm()->getDataObject()->getTypeInstance()->hasWeight()) {
>             $this->weightSwitcher->setValue(WeightResolver::HAS_WEIGHT);
>         }

Comment (engcom-Echo) [reply app/code/Magento/Catalog/Block/Adminhtml/Product/Helper/Form/Weight.php]: This check should be

Comment (Bartlomiejsz) [reply app/code/Magento/Catalog/Block/Adminhtml/Product/Helper/Form/Weight.php]: You're right @engcom-Echo, sorry for that. I restored this check

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
Failed functional tests not related to the changes in this PR

**Pull request merged**